reporting.txt

```diff
@@ -0,0 +1,952 @@
+   document, are to be interpreted as described in [RFC2119].
+
+   We also define the following terms for further use in this document:
+
+   o  STS Policy: A definition of the expected TLS availability and
+      behavior, as well as the desired actions for a given domain when a
+      sending MTA encounters different results.
+
+   o  Policy Domain: The domain against which an STS Policy is defined.
+
+   o  Sending MTA: The MTA initiating the delivery of an email message.
+
+2.  Related Technologies
+
+   o  This document is intended as a companion to the specification for
+      SMTP MTA Strict Transport Security (MTA-STS, TODO: Add ref).
+
+   o  The Public Key Pinning Extension for HTTP [RFC7469] contains a
+      JSON-based definition for reporting individual pin validation
+      failures.
+
+   o  The Domain-based Message Authentication, Reporting, and
+      Conformance (DMARC) [RFC7489] contains an XML-based reporting
+      format for aggregate and detailed email delivery errors.
+
+
+
+
+
+
+Margolis, et al.        Expires October 20, 2016                [Page 3]
+
+Internet-Draft                 SMTP-TLSRPT                    April 2016
+
+
+3.  Reporting Policy
+
+   SMTP TLSRPT policies are distributed via DNS from the Policy Domain's
+   zone, either through a new resource record, or as TXT records
+   (similar to DMARC policies) under the name "_smtp_tlsrpt".  (Current
+   implementations deploy as TXT records.)  For example, for the Policy
+   Domain "example.com", the recipient's SMTP STS policy can be
+   retrieved from "_smtp_tlsrpt.example.com".
+
+   (Future implementations may move to alternate methods of policy
+   discovery or distribution.  See the section _Future_ _Work_ for more
+   discussion.)
+
+   Policies consist of the following directives:
+
+   o  "version": This value MUST be equal to "TLSRPT1".
+
+   o  "aggregate-report-uri": A URI specifying the endpoint to which
+      aggregate information about policy failures should be sent (see
+      the section _Reporting Schema_ for more information).  Two URI
+      schemes are supported: "mailto" and "https".
+
+      *  In the case of "https", reports should be submitted via POST
+         ([RFC2818]) to the specified URI.
+
+      *  In the case of "mailto", reports should be submitted to the
+         specified email address.  When sending failure reports via
+         SMTP, sending MTAs MUST NOT honor SMTP STS or DANE TLSA
+         failures.
+
+   o  "detailed-report-uri": Future use.  (There may also be a need for
+      enabling more detailed "forensic" reporting during initial stages
+      of a deployment.  To address this, the authors consider the
+      possibility of an optional additional "forensic reporting mode" in
+      which more details--such as certificate chains and MTA banners--
+      may be reported.  See the section _Future_ _Work_ for more
+      details.)
+
+3.1.  Example Reporting Policy
+
```

Comment:
We're trying to be more like HPKP reporting, which is more verbose. Since it doesn't have to go in DNS, I favor being more explicit (verbose) in the tag names.